icefoganalytics/safety
# Caching static files for offline usage

## Context
Website needs to function as a PWA. For this to work static files need to be cached to ensure visability and functionality of the site while offline.

## Changes 
- Static files are cached via serviceWorker 
- Caching is all done through workbox functions
- Service worker is "compiled" for build via `vite-plugin-pwa`

**NOTE** serviceWorker is NOT registered in development environment
**NOTE** `web/src/registorSW.js` is no longer used

## How to test
1. Build vite web service
```bash
cd web
npm run build
```
3. Serve site statically
```bash
cd app/dist/web
python3 -m http.server 8080 // api calls may break, see web/src/config.ts
```
4. Go to `localhost:8080`
5. [Go offline via dev tools](https://developer.chrome.com/docs/devtools/network/reference#offline)
6. Refresh page and confirm functionality
